zig-demo-app/app.zig

```diff
@@ -0,0 +1,174 @@
+// *********************************************************************
+// * ConfD action example written in Zig
+// *
+// * (C) 2021 Tail-f Systems
+// * Permission to use this code as a starting point hereby granted
+// * This is ConfD Sample Code.
+// *
+// * See the README file for more information
+// ********************************************************************/
+
+const std = @import("std");
+const c_imp = @cImport({
+    @cInclude("string.h");
+    @cInclude("poll.h");
+    @cInclude("confd_lib.h");
+    @cInclude("confd_dp.h");
+    @cInclude("confd.h");
+});
+const dm = @cImport(@cInclude("datamodel.h"));
+
+// Set the log level to info
+pub const log_level: std.log.Level = .info;
+
+//pub const NULL = @import("std").zig.c_translation.cast(?*anyopaque, @as(c_int, 0));
+pub const CONFD_ADDR: [*c]const u8 = "127.0.0.1";
+pub const EXAMPLE_APP_NAME: [*c]const u8 = "app_example";
+pub var ctlsock: c_int = undefined;
+pub var workersock: c_int = undefined;
+pub var dctx: [*c]c_imp.struct_confd_daemon_ctx = undefined;
+
+pub fn initAction(arg_uinfo: [*c]c_imp.struct_confd_user_info) callconv(.C) c_int {
+    std.log.info("==> initAction", .{});
+    var rv: c_int = c_imp.CONFD_OK;
+    var uinfo = arg_uinfo;
+
+    c_imp.confd_action_set_fd(uinfo, workersock);
+    std.log.info("<== initAction rv={}", .{rv});
+    return rv;
+}
+
+pub fn traceConfdKp(txt: []const u8, kp: [*c]c_imp.confd_hkeypath_t) void {
+    if (kp != null) {
+        var buf: [1024]u8 = undefined;
+        _ = c_imp.confd_pp_kpath(@ptrCast([*c]u8, &buf), buf.len, kp);
```

Comment:
I don't know anything about zig, what does this construct mean? Explicitly ignore return value?

Reply:
Yes, without underscore one gets compilation error.

```
./app.zig:44:29: error: expression value is ignored
        c_imp.confd_pp_kpath(@ptrCast([*c]u8, &buf), buf.len, kp);
```